src/Castle.Facilities.AspNetCore/CastleWindsorRegistrationExtensions.cs

```diff
@@ -53,9 +54,26 @@ public static void AddCastleWindsor(this IServiceCollection services, IWindsorCo
 		/// <param name="container">Windsor container to register framework types in</param>
 		public static void UseCastleWindsor<TTypeAssembly>(this IApplicationBuilder app, IWindsorContainer container)
 		{
-			container.Register(Classes.FromAssemblyInThisApplication(typeof(TTypeAssembly).Assembly).BasedOn<Controller>().LifestyleScoped());
-			container.Register(Classes.FromAssemblyInThisApplication(typeof(TTypeAssembly).Assembly).BasedOn<ViewComponent>().LifestyleTransient());
-			container.Register(Classes.FromAssemblyInThisApplication(typeof(TTypeAssembly).Assembly).BasedOn<TagHelper>().LifestyleTransient());
+			var assembly = typeof(TTypeAssembly).Assembly;
+			UseCastleWindsor(container, assembly);
+		}
+
+		/// <summary>
+		/// Use this to register all Controllers, ViewComponents and TagHelpers. Will register the entry assembly and its referenced items
+		/// </summary>
+		/// <param name="app">Application builder retained as extension</param>
+		/// <param name="container">Windsor container to register framework types in</param>
+		public static void UseCastleWindsor(this IApplicationBuilder app, IWindsorContainer container)
+		{
+			var assembly = Assembly.GetEntryAssembly();
```

Comment:
I would have thought this would use GetCallingAssembly (marked with no inlining) like Windsor's fluent registration API, not GetEntryAssembly.

This project in the solution is built against .NET Standard 2.0 not 1.6 so the method is back.

Reply:
Thanks for your reply :-)
I have previously tested a lot with entry vs calling. Entry is the correct one.
Then e.g. the project with the bootstrapped referencing this package can be in another project than the startup.
The point of the method to me. Register for the entry/startup assembly / process. The entry assembly is the culmination of everything and the top level. Thoughts?

Reply:
I'm happy to take yours and @fir3pho3nixx lead, I just wanted to make sure it was a fully conscious decision.

Reply:
I perfectly agree. I am happy for decision. Comments are welcome.
I will change if you guys agree it should be different.
I have had many issues with the get assembly calls. But recently most luck with aspnetcore and entry.
Any other experiences?

Reply:
I think this should be fine for now. I tested this with a published app and in visual studio and it worked.